hibernate-core/src/main/java/org/hibernate/type/descriptor/java/OffsetDateTimeJavaType.java

```diff
@@ -79,6 +82,29 @@ protected <X> TemporalJavaType<X> forTimestampPrecision(TypeConfiguration typeCo
 		return (TemporalJavaType<X>) this;
 	}
 
+	@Override
+	public int extractHashCode(OffsetDateTime value, SqlTypedMapping mapping, SessionFactoryImplementor sessionFactory) {
+		return extractHashCode( getValue( value, mapping, sessionFactory ) );
+	}
+
+	@Override
+	public boolean areEqual(OffsetDateTime one, OffsetDateTime another, SqlTypedMapping mapping, SessionFactoryImplementor sessionFactory) {
+		return areEqual(
+				getValue( one, mapping, sessionFactory ),
+				getValue( another, mapping, sessionFactory )
+		);
+	}
+
+	@Override
+	public OffsetDateTime getValue(OffsetDateTime value, SqlTypedMapping mapping, SessionFactoryImplementor sessionFactory) {
+		return DateTimeUtils.roundToPrecision(
+				value,
+				mapping.getPrecision() != null
+						? mapping.getPrecision()
+						: sessionFactory.getJdbcServices().getDialect().getDefaultTimestampPrecision()
+		);
+	}
+
```

Comment:
may be `mapping.getPrecision()` can return  `sessionFactory.getJdbcServices().getDialect().getDefaultTimestampPrecision() into the mapping.getPrecision();` when no precision has been set;